feat(i18n): refined locales (#9200)
withastro/astro: .changeset/fluffy-dolls-sleep.md

---
'astro': minor
---

> Add a new way to configure the `i18n.locales` array.

Reviewer: ```suggestion
Adds a new way to configure the `i18n.locales` array.
```

> Developers can now assign a custom path that can span multiple language codes:

Reviewer 2: ```suggestion
Developers can now assign a custom URL path prefix that can span multiple language codes:
```

```js
// astro.config.mjs
export default defineConfig({
  experimental: {
      i18n: {
          defaultLocale: "english",
          locales: [

>             { path: "ensligh", codes: ["en", "en-US"]}

Reviewer: ```suggestion
            { path: "english", codes: ["en", "en-US"]}
```

>           locales: [
>             { path: "english", codes: ["en", "en-US"]}
>           ],

Reviewer 2: ```suggestion
          locales: [
            "de",
            { path: "english", codes: ["en", "en-US"]},
            "fr",
          ],
```
If this is actually true ( :smile: ), then it could be helpful to show the mix of the two formats together?

If not true, and all `locales` entries must be of the same type, I wouldn't bother to actually explain that here, but I would show a second language using the same format to imply that, and nudge people to using one configuration or the other.

          routingStrategy: "prefix-always"
      }
  }
})
```

> With this setting, the URL of the default locale will be start with `/english`. When computing `Astro.preferredLocale`, Astro will use the `codes`.

Reviewer 2: ```suggestion
With the above configuration, the URL prefix of the default locale will be `/english/`. When computing `Astro.preferredLocale`, Astro will use the `codes`.
```
